### PR TITLE
[rllib] Add debug info back to PPO and fix optimizer compatibility

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Make it work with sync samples optimizer too (this enables LSTM), and add back debug stats.

### CODE REVIEW

File: python/ray/rllib/optimizers/multi_gpu_optimizer.py [hunk 1]
```diff
@@ -116,13 +120,12 @@ def step(self, postprocess_fn=None):
                         self.sess,
                         permutation[batch_index] * self.per_device_batch_size)
                     for k, v in batch_fetches.items():
-                        iter_extra_fetches[k] += [v]
-                for k, v in iter_extra_fetches.items():
-                    all_extra_fetches[k] += [v]
+                        iter_extra_fetches[k].append(v)
+                print(i, _averaged(iter_extra_fetches))
```
Comment (ericl) [inline python/ray/rllib/optimizers/multi_gpu_optimizer.py]: This was incorrect before; you want to return the last epoch's stats, not the mean across all epochs.

**Review by richardliaw — COMMENTED** [review]
(comments were made from partial review a couple days ago)

File: python/ray/rllib/optimizers/multi_gpu_optimizer.py [hunk 2]
```diff
@@ -94,18 +95,21 @@ def step(self, postprocess_fn=None):
...
         with self.load_timer:
             tuples_per_device = self.par_opt.load_data(
                 self.sess,
                 samples.columns([key for key, _ in self.policy.loss_inputs()]))
 
         with self.grad_timer:
-            all_extra_fetches = defaultdict(list)
             num_batches = (
                 int(tuples_per_device) // int(self.per_device_batch_size))
+            print("== sgd epochs ==")
```
[the remark this answers is not in the document]
Comment (richardliaw) [reply python/ray/rllib/optimizers/multi_gpu_optimizer.py]: (flag off all of the prints in the optimizers with `verbose`?)

File: python/ray/rllib/evaluation/postprocessing.py [hunk 3]
```diff
@@ -44,6 +44,7 @@ def compute_advantages(rollout, last_r, gamma, lambda_=1.0, use_gae=True):
         rewards_plus_v = np.concatenate(
             [rollout["rewards"], np.array([last_r])])
         traj["advantages"] = discount(rewards_plus_v, gamma)[:-1]
+        traj["value_targets"] = np.zeros_like(traj["advantages"])
```
Comment (richardliaw) [inline python/ray/rllib/evaluation/postprocessing.py]: You can use a critic without using GAE, but this does not allow that functionality. Can you add documentation noting this?

File: python/ray/rllib/optimizers/multi_gpu_optimizer.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/optimizers/multi_gpu_optimizer.py]: Eh, I'm not sure why returning all epoch stats makes sense in any scenario, unless your epochs were very small and the values noisy.

File: python/ray/rllib/optimizers/multi_gpu_optimizer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/optimizers/multi_gpu_optimizer.py]: Let's do this with log level in the future. For now, we've always printed by default so this is just restoring 0.4 functionality.

File: python/ray/rllib/evaluation/postprocessing.py (hunk 3 above)
Comment (ericl) [reply python/ray/rllib/evaluation/postprocessing.py]: Done

**Review by richardliaw — APPROVED** [review]

**Pull request merged**